packages/backend/src/scheduler/SchedulerTask.ts

```diff
@@ -746,6 +758,180 @@ export const sendEmailNotification = async (
+            target: gdriveId,
+            targetType: 'gdrive',
+            status: SchedulerJobStatus.STARTED,
+        });
+
+        // Backwards compatibility for old scheduled deliveries
+        const { url, details, pageType, imageUrl, csvUrl, csvUrls } =
+            notification.page ??
+            (await getNotificationPageData(scheduler, jobId));
+        const schedulerUrl = `${url}?scheduler_uuid=${schedulerUuid}`;
+
+        if (format === SchedulerFormat.IMAGE) {
+            throw new Error('Not implemented');
+        } else if (savedChartUuid) {
+            if (csvUrl === undefined) {
+                throw new Error('Missing CSV URL');
+            }
+
+            const csvContent = await fsPromise.readFile(csvUrl.localPath, {
+                encoding: 'utf-8',
+            });
+
+            const refreshToken = await userService.getRefreshToken(
+                scheduler.createdBy,
+            );
+            await googleDriveClient.appendToSheet(
+                refreshToken,
+                gdriveId,
+                csvContent,
+            );
+        } else if (dashboardUuid) {
+            throw new Error('Not implemented');
+        } else {
+            throw new Error('Not implemented');
+        }
+
+        analytics.track({
+            event: 'scheduler_notification_job.completed',
+            anonymousId: LightdashAnalytics.anonymousId,
+            properties: {
+                jobId,
+                schedulerId: schedulerUuid,
+                schedulerTargetId: schedulerGdriveTargetUuid,
+                type: 'gdrive',
+                format,
+                resourceType:
+                    pageType === LightdashPage.CHART ? 'chart' : 'dashboard',
+            },
+        });
+        schedulerService.logSchedulerJob({
+            task: 'sendGdriveNotification',
+            schedulerUuid,
+            jobId,
+            jobGroup: notification.jobGroup,
+
+            scheduledTime,
+            target: gdriveId,
+            targetType: 'gdrive',
+            status: SchedulerJobStatus.COMPLETED,
+        });
+    } catch (e) {
+        analytics.track({
+            event: 'scheduler_notification_job.failed',
+            anonymousId: LightdashAnalytics.anonymousId,
+            properties: {
+                error: `${e}`,
+                jobId,
+                schedulerId: schedulerUuid,
+                schedulerTargetId: schedulerGdriveTargetUuid,
+                type: 'gdrive',
+            },
+        });
+        schedulerService.logSchedulerJob({
+            task: 'sendGdriveNotification',
+            schedulerUuid,
+            jobId,
+            jobGroup: notification.jobGroup,
+            scheduledTime,
+            targetType: 'gdrive',
+            status: SchedulerJobStatus.ERROR,
+            details: { error: e.message },
+        });
+
+        throw e; // Cascade error to it can be retried by graphile
+    }
+};
+
+const logScheduledTarget = async (
```

Comment:
refactor this logic, now we have 1 more target we can't do `isSlackTarget(target)) ? slack : email `